### PR TITLE
Issue #165 Support for CentOS v5 boxes 

### PR DESCRIPTION
**Pull request by khaije1**
Minor changes to enable Linux CentOS v5 to work correctly with Landrush.

Tested OK on CentOS versions 5, 6, 7

### CODE REVIEW

**Comment by hferentschik** [comment]
thanks @khaije1 for the pull request. Looking into it asap.

> Tested OK on CentOS versions 5, 6, 7

How does this effect other non CentOS Linux versions?

**Comment by khaije1** [comment]
All the tools and paths are generally stable across distros. 
In particular, the `ip` utility has been around for a long long time and it's presence should be a safe assumptions. I'm not thrilled about explicitly declaring the path to iptables, (or even relying on it where firewalld is of some prominence), but it's doesn't do anything to worsen or diminish the way these situation are handled.

I'll plan to test on a variety of debian and fedora -based distros. Any particular suggestions?

File: lib/landrush/cap/linux/read_host_visible_ip_address.rb [hunk 1]
```diff
@@ -39,7 +39,7 @@ def self.read_host_visible_ip_address(machine)
         end
 
         def self.command
-          %Q(hostname -I)
+          %Q(hostname -I 2>/dev/null || /sbin/ip ad | sed -ne 's/^\\W*inet \\([0-9.]*\\).* global .*$/\\1/p' 2>/dev/null)
```
Comment (hferentschik) [inline lib/landrush/cap/linux/read_host_visible_ip_address.rb]: You cannot really assume _ip_ is installed. This was a long standing discussion. It is not. The whole of _read_host_visible_ip_address_ will be changed by pull request #193. We will be introducing a new guest capability which will copy a binary onto the guest which will determine the available interfaces.

File: lib/landrush/cap/linux/configured_dns_servers.rb [hunk 2]
```diff
@@ -4,9 +4,9 @@ module Linux
       module ConfiguredDnsServers
         def self.configured_dns_servers(machine)
           return @dns_servers if @dns_servers
-          machine.communicate.sudo('cat /etc/resolv.conf | grep ^nameserver') do |type, data|
+          machine.communicate.sudo('sed -ne \'s/^nameserver \([0-9.]*\)$/\1/p\' /etc/resolv.conf') do |type, data|
```
Comment (hferentschik) [inline lib/landrush/cap/linux/configured_dns_servers.rb]: So why this change. What is the actual issue you are addressing? The _cat_ command did return something wrong on CentOS? I

File: lib/landrush/cap/linux/add_iptables_rule.rb [hunk 3]
```diff
@@ -3,7 +3,7 @@ module Cap
     module Linux
       module AddIptablesRule
         def self.add_iptables_rule(machine, rule)
-          _run(machine, %Q(iptables -C #{rule} 2> /dev/null || iptables -A #{rule}))
+          _run(machine, %Q(/sbin/iptables -C #{rule} || /sbin/iptables -A #{rule}))
```
Comment (hferentschik) [inline lib/landrush/cap/linux/add_iptables_rule.rb]: So why the full path?

**Comment by hferentschik** [comment]
> I'll plan to test on a the oldest supported and the most recent versions of debian, arch, coreos, ubuntu, and fedora. Any particular suggestions?

All :-) But seriously, using the cucumber test framework we already have - https://github.com/vagrant-landrush/landrush/blob/master/features/dns_resolution.feature

If you keep adding boxes to the [example list](https://github.com/vagrant-landrush/landrush/blob/master/features/dns_resolution.feature#L23) you can verify the DNS setup for all machine types you like. This will of course take some time and it is probably not something we want to have enabled all the time, but one can add them to see how it goes and then comment everything out except the most popular ones.

File: lib/landrush/cap/linux/read_host_visible_ip_address.rb (hunk 1 above)
Comment (khaije1) [inline lib/landrush/cap/linux/read_host_visible_ip_address.rb]: This PR can be considered complimentary to the binary solution proposed in #193.

The `ip` utility was mentioned [here](https://github.com/vagrant-landrush/landrush/pull/193#issuecomment-225037763), but I dont see any information designating it as an unsuitable for this purpose. According to the man page it was added alongside linux 2.2 in 1999.  On Linux, `ip` is the long-term replacement for `ifconfig`, so is a better option in all respects.

File: lib/landrush/cap/linux/add_iptables_rule.rb (hunk 3 above)
Comment (khaije1) [inline lib/landrush/cap/linux/add_iptables_rule.rb]: The default path on CentOS v5 is "/usr/kerberos/bin:/usr/local/bin:/bin:/usr/bin".

File: lib/landrush/cap/linux/configured_dns_servers.rb (hunk 2 above)
Comment (khaije1) [inline lib/landrush/cap/linux/configured_dns_servers.rb]: This change is basically unimportant.

Comment (hferentschik) [inline lib/landrush/cap/linux/configured_dns_servers.rb]: ok

File: lib/landrush/cap/linux/read_host_visible_ip_address.rb (hunk 1 above)
Comment (hferentschik) [inline lib/landrush/cap/linux/read_host_visible_ip_address.rb]: I merged the landrush-ip integration now. This change is obsolete.

**Comment by hferentschik** [comment]
@khaije1, I merged pull request #193. Would you mind verifying latest master with CentOS guests? Let's check what changes are still required (if any).

**Comment by khaije1** [comment]
I'm stuck behind https://github.com/mitchellh/vagrant/issues/6775, but will test at next opportunity.

**Comment by khaije1** [comment]
Will test/verify at next opportunity.

**Comment by hferentschik** [comment]
@khaije1 any update on this?

**Comment by khaije1** [comment]
Unfortunately I can only report that the latest v1 release, it seems #193 is in https://github.com/vagrant-landrush/landrush/tree/v1.0.0 ... is that correct? In my recent testing v1.0.0 isn't working on CentOS version 5, 6, or 7. I'd like to dig further into why but haven't had time to. 

The VM's will now progress as normal but once booted, DNS resolution is broken for landrush-contrived A records and for public DNS.

I've switched to using #191 which is working well with vagrant v1.8.5 and vagrant-vmware-fusion 4.0.10, in order to be able to continue with my day-to-day work.

**Comment by hferentschik** [comment]
Closing this pull request, since it needs more work. Happy to look at an updated pull request.
